sklearn/linear_model/_glm/tests/test_glm.py

```diff
@@ -291,8 +282,9 @@ def test_glm_regression_hstacked_X(solver, fit_intercept, glm_dataset):
     model.fit(X, y)
 
     rtol = 1e-4
-    assert model.intercept_ == pytest.approx(intercept, rel=rtol)
-    assert_allclose(model.coef_, np.r_[coef, coef], rtol=rtol)
+    atol = 1e-6
+    assert model.intercept_ == pytest.approx(intercept, rel=rtol, abs=atol)
+    assert_allclose(model.coef_, np.r_[coef, coef], rtol=rtol, atol=atol)
```

Comment:
Note: that I had to add quite a largish `atol` for this test to pass with the new init. Otherwise the tests would fail for a few random seeds, either with wide or long datasets for various distribution families.